### PR TITLE
Add regressor result

### PR DESCRIPTION
**Pull request by liam-sbhoo**
### Change Description

**Brief** (a few bullet points describing your changes, use full sentences and try to link lines in the code whenever needed)

- create a helper class `RegressionPredictResult` to handle:
    - keep important info of `TabPFNRegressor().predict_full(...)`
    - serialize and deserialize (to be used by server and client)
 - add respective test  

**Details** (add details if your pull request is more complicated and harder to understand from the code alone)

### Standard Qs (leave questions that do not apply blank)
**If you broke behavior: Please describe what behavior you broke and how you inform people to not get stuck trying to use the old behavior.**

No

**If you used new dependencies: Did you add them to `requirements.txt`?**

No new dependencies

**Who did you ping on Mattermost to review your PR? Please ping that person again whenever you are ready for another review.**

@SamuelGabriel 

---
*Please do not mark comments/conversations as resolved unless you are the assigned reviewer. This helps maintain clarity during the review process.*

### CODE REVIEW

File: regression_pred_result.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        # assert all values are of the same type
+        for val in [self.mean, self.median, self.mode, *self.quantiles.values()]:
+            assert isinstance(val, self._val_type)
+
+    @property
+    def val_type(self):
+        return self._val_type
+
+    @staticmethod
+    def serialize(res: "RegressionPredictResult") -> {str: list}:
```
Comment (SamuelGabriel) [inline regression_pred_result.py:50]: i think serialize is a slightly bad name here, as that usually implies to str. Maybe something like to `to_basic_representation` and `from_basic_representation`?

File: regression_pred_result.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
+import numpy as np
+import torch
+
+
+class RegressionPredictResult:
+    def __init__(self, res: {}):
+        # res:
```
Comment (SamuelGabriel) [inline regression_pred_result.py:7]: please get rid of this comment

**Review by SamuelGabriel — APPROVED** [review]
This looks good to me, I would just rename one function. Thanks :)

File: regression_pred_result.py (hunk 1 above)
Comment (liam-sbhoo) [reply regression_pred_result.py]: Yup, I used `to_basic_representation` and `from_basic_representation` as per suggested :)

**Pull request merged**